File: gdx2d-demoDesktop/src/ch/hevs/gdx2d/demos/image_drawing/DemoSimpleImage.java
package ch.hevs.gdx2d.demos.image_drawing;

import hevs.gdx2d.components.bitmaps.BitmapImage;
import hevs.gdx2d.lib.GdxGraphics;
import hevs.gdx2d.lib.PortableApplication;

/**
 * Demo application to demonstrate how to load a file which
 * should be located in a {@code data} folder at the root
 * of the project.
 *
 * @author Pierre-André Mudry (mui)
 * @version 1.0
 */
public class DemoSimpleImage extends PortableApplication {
	float counter = 0;
	BitmapImage imgBitmap;

	public static void main(String[] args) {
		new DemoSimpleImage();
	}

	@Override
	public void onInit() {
		// Sets the window title
		setTitle("Simple image drawing, mui 2013");

		// Loads the image that will be displayed in the middle of the scren
		imgBitmap = new BitmapImage("data/images/compass_150.png");
	}

	@Override
	public void onGraphicRender(GdxGraphics g) {
		/**
		 * Rendering
		 */
		// Clear the screen
		g.clear();

		// Render an image which turns with scale
		g.drawPicture(getWindowWidth() / 2, getWindowHeight() / 2, imgBitmap);

		g.drawFPS();        // Draws the number of frame per second
		g.drawSchoolLogo(); // Draws the school logo
	}
}

File: gdx2d-demoDesktop/src/main/scala/ch/hevs/gdx2d/demos/image_drawing/DemoSimpleImage.scala
package ch.hevs.gdx2d.demos.image_drawing

import ch.hevs.gdx2d.components.bitmaps.BitmapImage
import ch.hevs.gdx2d.desktop.DesktopApplication
import ch.hevs.gdx2d.lib.GdxGraphics

/**
 * Demo application to demonstrate how to load a file which should be located
 * in a `data` folder at the root of the project.
 *
 * @author Pierre-André Mudry (mui)
 * @version 2.0
 */
class DemoSimpleImage extends DesktopApplication {

  protected var counter: Float = 0f
  protected var imgBitmap: BitmapImage = _

  override def onInit(): Unit = {
    setTitle("Simple image drawing, mui 2013")
    imgBitmap = new BitmapImage("images/compass_150.png")
  }

  override def onGraphicRender(g: GdxGraphics): Unit = {
    g.clear()
    g.drawPicture((getWindowWidth / 2).toFloat, (getWindowHeight / 2).toFloat, imgBitmap)
    g.drawFPS()
    g.drawSchoolLogo()
  }
}

object DemoSimpleImage {
  def main(args: Array[String]): Unit = new DemoSimpleImage().launch()
}
